src/pages/index.js

```diff
@@ -0,0 +1,318 @@
+    const modal = document.querySelector(".modal_opened");
+    if (modal) closeModal(modal);
+  }
+};
+
+// ------------------- SELECTORS -------------------
+
+const selectors = {
+  profileEditButton: document.querySelector(".profile__edit-btn"),
+  cardAddButton: document.querySelector(".profile__add-btn"),
+  profileName: document.querySelector(".profile__name"),
+  profileDescription: document.querySelector(".profile__description"),
+  profileForm: document.querySelector(".modal__form"),
+  nameInput: document.querySelector("#profile-name-input"),
+  jobInput: document.querySelector("#profile-description-input"),
+  editProfileModal: document.querySelector("#edit-profile-modal"),
+  cardModal: document.querySelector("#add-card-modal"),
+  cardForm: document.querySelector("#add-card-form"),
+  cardTemplate: document.querySelector("#card-template").content,
+  cardsContainer: document.querySelector(".cards"),
+  previewModal: document.querySelector("#preview-modal"),
+  previewImage: document.querySelector(".modal__image"),
+  previewCaption: document.querySelector(".modal__caption"),
+  avatarModal: document.querySelector("#avatar-modal"),
+
+  avatarPic: document.querySelector("#avatar-pic"),
+};
+
+// ------------------- API INSTANCE -------------------
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d19feaee-a024-4f2d-b0a9-cad223e76830",
+    "Content-Type": "application/json",
+  },
+});
+
+// ------------------- FETCH & RENDER DATA -------------------
+api
+  .getAppInfo()
+  .then(([userInfo, cards]) => {
+    console.log("Cards fetched:", cards);
+    selectors.profileName.textContent = userInfo.name;
+    selectors.profileDescription.textContent = userInfo.about;
+    selectors.avatarPic.src = userInfo.avatar;
+    cards.forEach((card) => renderCard(card, "append"));
+  })
+  .catch(console.error);
+
+// ------------------- FORM HANDLERS -------------------
+const resetFormAndButton = (form) => {
+  form.reset();
+  const inputList = Array.from(form.querySelectorAll(settings.inputSelector));
+  const submitButton = form.querySelector(settings.submitButtonSelector);
+  toggleButtonState(inputList, submitButton, settings);
+  resetValidation(form, settings);
+};
+
+const handleAvatarFormSubmit = (evt) => {
+  evt.preventDefault();
+  const avatarInput = document.querySelector("#profile-avatar-input").value;
+  const saveButton = avatarForm.querySelector(".modal__submit-btn");
+  saveButton.textContent = "Saving...";
+
+  api
+    .updateUserAvatar({ avatar: avatarInput })
+    .then((updatedUserInfo) => {
+      selectors.avatarPic.src = updatedUserInfo.avatar;
+      closeModal(selectors.avatarModal);
+      resetFormAndButton(avatarForm);
+    })
+    .catch((err) => console.error("Error updating avatar:", err))
+    .finally(() => {
+      saveButton.textContent = "Save";
+      saveButton.disabled = true; // Disable the button until input is valid again
```

Comment:
It's not needed here because you have `resetFormAndButton` for that in line 112

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.

**Please, correct it everywhere in the code**